index.js

```diff
@@ -91,29 +91,58 @@ function init (ssb, config) {
 
   /* start listeners */
   const rebuildManager = new RebuildManager(ssb)
-  listen.addMember(ssb, m => {
-    const { root, groupKey } = m.value.content
-    ssb.get({ id: root, meta: true }, (err, groupInitMsg) => {
-      if (err) throw err
+  const processedNewAuthors = {}
+  pull(
+    listen.addMember(ssb),
+    pull.asyncMap((m, cb) => {
+      const { root, groupKey } = m.value.content
+      ssb.get({ id: root, meta: true }, (err, groupInitMsg) => {
+        if (err) throw err
 
-      const groupId = buildGroupId({ groupInitMsg, groupKey })
-      const authors = unique([
-        groupInitMsg.value.author,
-        m.value.author,
-        ...m.value.content.recps.filter(isFeed)
-      ])
+        const groupId = buildGroupId({ groupInitMsg, groupKey })
+        const authors = unique([
+          groupInitMsg.value.author,
+          m.value.author,
+          ...m.value.content.recps.filter(isFeed)
+        ])
+
+        const record = keystore.group.get(groupId)
+        // if we haven't been in the group since before, register the group
+        if (record == null) {
+          return keystore.group.register(groupId, { key: groupKey, root }, (err) => {
+            if (err) return cb(err)
+            processAuthors(cb)
+          })
+        } else {
+          processAuthors(cb)
+        }
 
-      keystore.processAddMember({ groupId, groupKey, root, authors }, (err, newAuthors) => {
```

Comment:
@Powersource the  bug I suspect is in here.
Note in this module:
- the `keystore` processing addition of new members does:
    - checking against keystore current state
    - UPDATING that internal membership state accordingly
- unboxing is handled by an `unboxer` function... this unboxer is installed with `ssb.addUnboxer` and importantly IT DEPENDS ON `keystore`

If you follow the thread up into `envelope.js` you'll see the unboxer asks the keystore which keys it should try - is this message by an author in this group? They try the group the key for decryption (otherwise don't bother)

Reply:
hmmm so we might not have to just remove uses of `keystore.group.*` functions but also `keystore.author.*` functions? completely missed that those existed lol

Reply:
fixed with 764200f5daca62f1839c73df4189158dce36851a Nice that I could just update a function in ssb-tribes (even though it took me some time to realize that) instead of somewhere deep in db/box code